### ISSUE TITLE
[Slider] onMouseDown useCallback Function is Not Being Updated

### ISSUE THREAD

**Issue by zvakanaka** [reach/reach-ui#741]
### Current Behavior

Passing a `useCallback` function to the `onMouseDown` prop (and the other similar props) keeps the original function around.

### Expected behavior

A `useCallback` function passed to the `onMouseDown` prop (and other similar props) should be updated when something in the dependency array changes.

### Reproducible example

[Sample of the issue in CodeSandbox](https://codesandbox.io/s/focused-shockley-l31t0)

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Reach Package    |    `@reach/slider`     |      `0.13.0`   |
| React            |    `react`     |      `17.0.1`   |
| Browser          |     Firefox    |   `86.0`      |
| Assistive tech   |     None    |      None   |
| Node             |    node     |      `v14.13.1`   |
| npm/yarn         |     yarn    |    `1.21.1`     |
| Operating System |    macOS     |     `10.15.7`    |

**Comment by chaance** [comment]
Good catch -- we're probably trying to be a little too clever here since by trying to avoid re-renders internally. That should probably be left to the user in this case. I've patched this up in a sandbox but am anticipating that you might have another problem unless you're trying to get the *previous* value in your event handler. Haven't merged anything yet because I need to test a few edge cases with these changes: https://codesandbox.io/s/intelligent-ives-0x6rz?file=/src/App.js

**Comment by chaance** [comment]
A note for self: this change will affect all pointer-related events, and we'll want to add notes in the documentation if users want to opt-in to the current behavior after the change.
